client/components/pages/files/FilesTable.vue

```diff
@@ -26,43 +45,107 @@ provide(FileTableKey, {
   selectedRow,
   selectedRowId,
   updateSelectedRow,
-  removeSelectedRow,
   unselectRow,
+  refresh,
 })
 
 const contextMenu = ref<InstanceType<typeof PContextMenu> | null>(null)
 provide(FileTableContextMenuKey, contextMenu)
+
+function resetPage() {
+  page.value = 0
+  pageSize.value = 10
+  pageDefinition.value = {}
+  refresh()
+}
+
+function onPage(event: DataTablePageEvent) {
+  page.value = event.page
+  pageSize.value = event.rows
+}
+
+function onSort(event: DataTableSortEvent) {
+  pageDefinition.value = {
+    ...pageDefinition.value,
+    sorter: event.multiSortMeta?.map((sort: DataTableSortMeta) => ({
```

Comment:
im now realizing we should just have a sortDefinition ref, (we will have a filterDefinition ref too later, or something akin to it).

and pageDefinition should just be a `computed(() => ({ sorter: sortDefinition, filter: filterDefinition }))`

This would avoid such things as spreading the object even when we know we will never write directly on pageDefinition semantically speaking.